### PR TITLE
create apikeys

### PR DESCRIPTION
**Pull request by phenolphtalein**
#203 
• added a button on the /users for admin to create api keys for all users who don't have one
• sends email to existing user after api key has been created
• added api key information on users/show page
• create api key when user signs up

### CODE REVIEW

**Comment by dlebauer** [comment]
* for new users, api key included in sign-up email

**Comment by gsrohde** [comment]
@phenolphtalein , @dlebauer 

I guess this gets the job done, but it seems a little odd to me to create a button in the Web interface that you are probably only ever going to use once.

**Comment by dlebauer** [comment]
@phenolphtalein 
* what is the purpose of the button? c
* an we generate api keys for existing users without sending an email?

**Comment by phenolphtalein** [comment]
I will change the button to a script then. For the email, I thought we'd want to notify the existing users that an api key has been created for them.

**Comment by dlebauer** [comment]
It would be nice to have a "generate api key" that will generate an api key and send to a user on a per-user basis.

That way, an individual user could change their apikey (e.g. if they accidentally compromised one by committing it to public code). And, we wouldn't need to generate an API key (and email) everyone on the list.

**Comment by phenolphtalein** [comment]
Updated the page and removed the button to create all apikeys. Added a button to generate a new key for each individual user.

**Comment by dlebauer** [comment]
Can you please document the rake task in the documentation? https://www.authorea.com/users/5574/articles/7062/_show_article

**Comment by dlebauer** [comment]
close #203 when deployed

File: app/views/contact_mailer/apikey_email.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h1>Hello, <%= @user[:name] %></h1>
+    <p>
+      Your apikey for BETYdb is <%@user[:apikey]%>
```
Comment (gsrohde) [inline app/views/contact_mailer/apikey_email.html.erb]: This should be `<%= user[:apikey] %>`.

File: app/views/contact_mailer/apikey_email.txt.erb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Hello, <%= @user[:name] %>
+
+Your apikey for BETYdb is <%@user[:apikey]%>
```
Comment (gsrohde) [inline app/views/contact_mailer/apikey_email.txt.erb]: This should be `<%= user[:apikey] %>`.

**Comment by gsrohde** [comment]
Please make fixes to the two email template files.

**Comment by phenolphtalein** [comment]
@gsrohde  Fixed.

**Comment by gsrohde** [comment]
@phenolphtalein  I have merged this, but I made some tweaks to DRY up the code and clean up the routes a bit.  (I also tweaked the e-mail templates a bit.)  If you're interested in seeing my changes, do
```
git diff 5de42f7 09b0b4d
```
and also look at the git log.

**Pull request merged**